core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java

```diff
@@ -250,14 +252,185 @@ In practice, the stop train will require the route in front of him (zone and swi
+    }
+
+    /*
+    Test that overtaking conflicts are correctly processed in the case of 2 trains, one is
+    stopping for 10 min, the other is direct.
+    For all combinations of:
+    - stop with reception on stop signal or not
+    - trains being close (5min) or far (1hour) from each other
+    - first train leaving being the one with stop (overtaking) or the direct (no overtake)
+     */
+    @ParameterizedTest
+    @CsvSource({
+        "false, 300, true, true",
+        "true, 300, false, false",
+        "false, 3600, false, false",
+        "true, 3600, false, false",
+        "false, -300, false, false",
+        "true, -300, false, false",
+        "false, -3600, false, false",
+        "true, -3600, false, false",
+    })
+    public void conflictDetectionForOvertakeInStation(
+            boolean onStopSignal,
+            double directStartTime, // stopping train starts at 0
+            boolean hasRoutingConflict,
+            boolean hasSpacingConflict)
+            throws Exception {
+        var rjsInfra = Helpers.getExampleInfra("small_infra/infra.json");
+        var fullInfra = fullInfraFromRJS(rjsInfra);
+        var rawInfra = fullInfra.rawInfra();
+
+        var ta6 = getTrackSectionFromNameOrThrow("TA6", rawInfra);
+        var td0 = getTrackSectionFromNameOrThrow("TD0", rawInfra);
+
+        var chunkPathCenter = chunkPathFromRoutes(
+                rawInfra,
+                List.of("rt.DA0->DA5", "rt.DA5->DC5", "rt.DC5->DD2"),
+                makeTrackLocation(ta6, fromMeters(1000)), // start after DA3
+                makeTrackLocation(td0, fromMeters(24820)));
+        var pathPropsCenter = makePathProperties(rawInfra, chunkPathCenter, null);
+        var chunkPathNorth = chunkPathFromRoutes(
+                rawInfra,
+                List.of("rt.DA0->DA5", "rt.DA5->DC4", "rt.DC4->DD2"),
+                makeTrackLocation(ta6, fromMeters(1000)),
+                makeTrackLocation(td0, fromMeters(24820)));
+        var pathPropsNorth = makePathProperties(rawInfra, chunkPathNorth, null);
+
+        var stop = new TrainStop(9700, 600, onStopSignal);
+        var simResultCenterWithStop =
+                simpleSim(fullInfra, pathPropsCenter, chunkPathCenter, 0, Double.POSITIVE_INFINITY, List.of(stop));
+        var simResultNorthOvertaking =
+                simpleSim(fullInfra, pathPropsNorth, chunkPathNorth, 0, Double.POSITIVE_INFINITY, List.of());
+
+        var reqWithStop = convertRequirements(0L, 0.0, simResultCenterWithStop.train);
+        var reqOvertaking = convertRequirements(1L, directStartTime, simResultNorthOvertaking.train);
+        var conflicts = ConflictsKt.detectConflicts(List.of(reqWithStop, reqOvertaking));
+        assert (hasRoutingConflict == conflicts.stream().anyMatch((conflict) -> conflict.conflictType == ROUTING));
+        assert (hasSpacingConflict == conflicts.stream().anyMatch((conflict) -> conflict.conflictType == SPACING));
+    }
+
+    @ParameterizedTest
```

Comment:
This test seems very similar to the one above the above. First, it'd be nice to put them together. Then, a comment like for the others would help.

Reply:
Moved the test and commented in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/864991bab8e6de95f0b4b2231e4fadec85098d44